frontend/src/components/LaunchForm.tsx

```diff
@@ -0,0 +1,131 @@
+import React, {useEffect, useState} from "react";
+import {makeStyles} from "@material-ui/core/styles";
+import {DEFUALT_ADDRESS, DEFAULT_PORT} from "../consts"
+
+interface LaunchFormProps {
+    adress : string;
+    setAddress : (s : string) => void;
+    port : string;
+    setPort : (s : string) => void;
+    login : string, 
+    setLogin : (s : string) => void;
+    password : string;
+    setPassword : (s : string) => void;
+    setFormPassed : (b : boolean) => void;
+}
+
+function LaunchForm({adress, setAddress, port, setPort, login, setLogin, password, setPassword, setFormPassed} : LaunchFormProps)
+{
+    const classes = useStyles();
+
+    const changeAddress =  (event : React.ChangeEvent<HTMLInputElement>) => {
+        setAddress(event.target.value.length && event.target.value[0].trim() === '' ? 
+            event.target.value.slice(1) : event.target.value);
+    }
+
+    const changePort = (event : React.ChangeEvent<HTMLInputElement>)  => {
+        setPort(event.target.value.length && event.target.value[0].trim() === '' ? 
+            event.target.value.slice(1) : event.target.value);
+    }
+
+    const changeLogin =  (event : React.ChangeEvent<HTMLInputElement>) => {
+        setLogin(event.target.value.length && event.target.value[0].trim() === '' ? 
+            event.target.value.slice(1) : event.target.value);
+    }
+
+    const changePassword = (event : React.ChangeEvent<HTMLInputElement>)  => {
+        setPassword(event.target.value.length && event.target.value[0].trim() === '' ? 
+            event.target.value.slice(1) : event.target.value);
+    }
+
+    const submitForm = (e : any) => {
+        e.preventDefault();
+
+        if (adress === "")
+            setAddress(DEFUALT_ADDRESS);
+
+        if (port == "")
+            setPort(DEFAULT_PORT);
+
+        //console.log(adress, port, login, password)
```

Comment:
co tu robi ten komentarz?

Reply:
Był do testowanie, potencjalnie jak bedziemy dodawac dodatkowe ficzery jak np. hashowanie hasła, dodatkowe wymagani na login/hasło/adres to moze sie przydac, ew mozna go usunać.